src/main/java/logging/Logger.java

```diff
@@ -146,17 +166,22 @@ public static String getLogDir() {
      */
     public static void setDebugOn() {
         debug = true;
+        logger = new Logger();
```

Comment:
No, its the other way around: A logger should run in its own thread. Therefore a Logger instance must be instantiated and managed by the Logger class. Now the static methods just provide a means to communicate with the logger thread.